contracts/contracts/strategies/CurvePoolBooster.sol

```diff
@@ -0,0 +1,334 @@
+        uint256 maxRewardPerVote,
+        address[] calldata blacklist,
+        uint256 bridgeFee,
+        uint256 additionalGasLimit
+    ) external onlyGovernorOrStrategist {
+        require(campaignId == 0, "Campaign already created");
+        require(numberOfPeriods > 1, "Invalid number of periods");
+        require(maxRewardPerVote > 0, "Invalid reward per vote");
+
+        // Handle fee (if any)
+        uint256 balanceSubFee = _handleFee();
+
+        // Approve the balanceSubFee to the campaign manager
+        IERC20(rewardToken).safeApprove(campaignRemoteManager, balanceSubFee);
+
+        // Create a new campaign
+        ICampaignRemoteManager(campaignRemoteManager).createCampaign{
+            value: bridgeFee
+        }(
+            ICampaignRemoteManager.CampaignCreationParams({
+                chainId: targetChainId,
+                gauge: gauge,
+                manager: address(this),
+                rewardToken: rewardToken,
+                numberOfPeriods: numberOfPeriods,
+                maxRewardPerVote: maxRewardPerVote,
+                totalRewardAmount: balanceSubFee,
+                addresses: blacklist,
+                hook: address(0),
+                isWhitelist: false
+            }),
+            targetChainId,
+            additionalGasLimit
+        );
+
+        emit BribeCreated(gauge, rewardToken, maxRewardPerVote, balanceSubFee);
+    }
+
+    /// @notice Manage the total reward amount of the campaign
+    /// @dev This function should be called after the campaign is created
+    /// @dev This will use all the token available in this contract
+    /// @param bridgeFee Fee to pay for the bridge
+    /// @param additionalGasLimit Additional gas limit for the bridge
+    function manageTotalRewardAmount(
+        uint256 bridgeFee,
+        uint256 additionalGasLimit
+    ) external onlyGovernorOrStrategist {
+        require(campaignId != 0, "Campaign not created");
+
+        // Handle fee (if any)
+        uint256 balanceSubFee = _handleFee();
+
+        // Approve the total reward amount to the campaign manager
+        IERC20(rewardToken).safeApprove(campaignRemoteManager, balanceSubFee);
+
+        // Manage the campaign
+        ICampaignRemoteManager(campaignRemoteManager).manageCampaign{
+            value: bridgeFee
+        }(
+            ICampaignRemoteManager.CampaignManagementParams({
+                campaignId: campaignId,
+                rewardToken: rewardToken,
+                numberOfPeriods: 0,
+                totalRewardAmount: balanceSubFee,
+                maxRewardPerVote: 0
+            }),
+            targetChainId,
+            additionalGasLimit
+        );
+
+        emit TotalRewardAmountUpdated(balanceSubFee);
+    }
+
+    /// @notice Manage the number of periods of the campaign
+    /// @dev This function should be called after the campaign is created
+    /// @param extraNumberOfPeriods Number of additional periods (cannot be 0)
```

Comment:
maybe add 
`param extraNumberOfPeriods Number of additional periods (cannot be 0) that will be added to already existing amount of periods`

Reply:
Done in this commit: https://github.com/OriginProtocol/origin-dollar/pull/2327/commits/49a88dd301b0bf05cdd01299a6b9104d2259e64a